skada/base.py

```diff
@@ -119,6 +119,25 @@ def transform(
         )
 
 
+class DAEstimator(BaseEstimator):
+    """ Generic DA estimator class
+
+    """
+
+    __metadata_request__fit = {'sample_domain': True}
+    __metadata_request__transform = {'sample_domain': True, 'allow_source': True}
```

Comment:
I guess we also need to specify request for `score`, `decision_function`, `predict_proba`

Reply:
Very good point  and one of the reasons why I want a class so that contributor do not need to worry about that ;)